### PR TITLE
Implemented snapshot stream support.

### PR DESCRIPTION
**Pull request by nscheibe**
Streams snapshots were not supported in snapshooter. Therefore we added some new json converters to serialize streams for our snapshots.

### CODE REVIEW

**Review by PascalSenn — APPROVED** [review]

File: src/Snapshooter/Core/Serialization/SnapshotSerializerSettings.cs [hunk 1]
```diff
@@ -1,60 +1,65 @@
...
+                    new StreamJsonConverter()
+                }
+        };
+
+    public abstract JsonSerializerSettings Extend(JsonSerializerSettings settings);
+
+    private class ChildFirstContractResolver : DefaultContractResolver
     {
-        public virtual int Order { get; } = 1;
+        static ChildFirstContractResolver() { Instance = new ChildFirstContractResolver(); }
```
Comment (PascalSenn) [inline src/Snapshooter/Core/Serialization/SnapshotSerializerSettings.cs:44]:
```suggestion
        public static ChildFirstContractResolver() { Instance = new ChildFirstContractResolver(); }
```

Comment (PascalSenn) [inline src/Snapshooter/Core/Serialization/SnapshotSerializerSettings.cs:44]: i would not create a static constructor for this

File: src/Snapshooter/Core/Serialization/SnapshotSerializerSettings.cs [hunk 2]
```diff
@@ -1,60 +1,65 @@
...
-                    properties = properties.OrderBy(p =>
-                    {
-                        IEnumerable<Type> d = ((Type)p.DeclaringType).BaseTypesAndSelf().ToList();
-                        return 1000 - d.Count();
-                    }).ToList();
-                }
-
-                return properties;
+                    IEnumerable<Type> d = ((Type)p.DeclaringType).BaseTypesAndSelf().ToList();
+                    return 1000 - d.Count();
```
Comment (PascalSenn) [inline src/Snapshooter/Core/Serialization/SnapshotSerializerSettings.cs:58]:
```suggestion
    return 1000 - ((Type)p.DeclaringType).BaseTypesAndSelf().Count();
```

Comment (nscheibe) [reply src/Snapshooter/Core/Serialization/SnapshotSerializerSettings.cs:58]: I completely agree

**Pull request merged**